tgui/packages/tgui/interfaces/PreferencesMenu/KeybindingsPage.tsx

```diff
@@ -439,37 +442,44 @@ export class KeybindingsPage extends Component<{}, KeybindingsPageState> {
                 return [
                   category,
                   <Stack key={category} vertical fill>
-                    {sortKeybindings(Object.entries(keybindings)).map(([keybindingId, keybinding]) => {
-                      const keys = this.state.selectedKeybindings![keybindingId] || [];
-
-                      const name = (
-                        <Stack.Item basis="25%">
-                          <KeybindingName keybinding={keybinding} />
-                        </Stack.Item>
-                      );
-
-                      return (
-                        <Stack.Item key={keybindingId}>
-                          <Stack fill>
-                            {name}
-
-                            {range(0, 3).map((key) => (
-                              <Stack.Item key={key} grow basis="10%">
-                                <KeybindingButton
-                                  currentHotkey={keys[key]}
-                                  typingHotkey={this.getTypingHotkey(keybindingId, key)}
-                                  onClick={this.getKeybindingOnClick(keybindingId, key)}
-                                />
+                    {sortKeybindings(Object.entries(keybindings))
+                      .filter(([keybindingId, keybinding]) => {
+                        return (
+                          !keybinding.pref_key ||
+                          data.character_preferences.game_preferences[keybinding.pref_key] === keybinding.pref_value
```

Comment:
what's the point in hiding the keybinds? It just makes it less discoverable imo.

Reply:
Because otherwise there is a massive wall of keybinds when only some of them can work depending on the mode. If you are using simplified bodyzone targeting there is no reason to display hotkeys that only work when you aren't using that mode. Having too many keybinds that don't do anything are equally less discoverable